criticality_score/run.py

```diff
@@ -460,41 +461,54 @@ def _worker(repo, param, return_dict):
     return result_dict
 
 
-def get_github_token_info(token_obj):
-    """Return expiry information given a github token."""
-    rate_limit = token_obj.get_rate_limit()
-    near_expiry = rate_limit.core.remaining < 50
-    wait_time = (rate_limit.core.reset - datetime.datetime.utcnow()).seconds
-    return near_expiry, wait_time
-
+def get_github_auth_token_info(auth_token):
+    """If near expiry, sets reset information given a github token."""
+    rate_limit = auth_token['token_obj'].get_rate_limit()
+    if rate_limit.core.remaining < 50:
+        auth_token['reset'] = rate_limit.core.reset
+        
 
 def get_github_auth_token():
     """Return an un-expired github token if possible from a list of tokens."""
     global _CACHED_GITHUB_TOKEN
-    global _CACHED_GITHUB_TOKEN_OBJ
-    if _CACHED_GITHUB_TOKEN_OBJ:
-        near_expiry, _ = get_github_token_info(_CACHED_GITHUB_TOKEN_OBJ)
-        if not near_expiry:
-            return _CACHED_GITHUB_TOKEN_OBJ
-
-    github_auth_token = os.getenv('GITHUB_AUTH_TOKEN')
-    assert github_auth_token, 'GITHUB_AUTH_TOKEN needs to be set.'
-    tokens = github_auth_token.split(',')
-
-    wait_time = None
-    token_obj = None
-    for token in tokens:
-        token_obj = github.Github(token)
-        near_expiry, wait_time = get_github_token_info(token_obj)
-        if not near_expiry:
-            _CACHED_GITHUB_TOKEN = token
-            _CACHED_GITHUB_TOKEN_OBJ = token_obj
-            return token_obj
-
-    print(f'Rate limit exceeded, sleeping till reset: {wait_time} seconds.',
-          file=sys.stderr)
-    time.sleep(wait_time)
-    return token_obj
+
+    # Initialize _CACHED_GITHUB_TOKENS that stores token, token_obj and reset data as a list
+    if not _CACHED_GITHUB_TOKENS:
+        github_auth_token = os.getenv('GITHUB_AUTH_TOKEN')
+        assert github_auth_token, 'GITHUB_AUTH_TOKEN needs to be set.' # TODO Validate earlier?
+        for token in github_auth_token.split(','):
+            token_obj = github.Github(token)
```

Comment:
We are recreating all token objects after every repo, this is not needed, also feels unnecessary to add reset value in there, since it needs to be rechecked. Also get_github_auth_token_info is weird now, it is just set the reset value, not getting any info.

If you want to just do the minimum wait time, you can do in the loop by capturing least value of wait_time while iterating in the loop. that is it, no other change needed.

    for token in tokens:	            
        token_obj = github.Github(token)
        near_expiry, wait_time = get_github_token_info(token_obj)
        if not near_expiry:	
            _CACHED_GITHUB_TOKEN = token	        
            _CACHED_GITHUB_TOKEN_OBJ = token_obj
            return token_obj

Reply:
Okay, only kept "min_wait_time" part and removed the other changes